Happy Thoughts By Anna

Anna2024WebDev opened the pull request:
## Netlify link

https://happythoughtsbyanna.netlify.app/

Anna2024WebDev committed:
added components for form, hearts and thought list and corresponding css files, fetched the thoughts data from API

Anna2024WebDev committed:
added function for heart likes and if/else statements for example min/max number of characters, also added styling to some components

Anna2024WebDev committed:
updated styling to button, form and textarea

Anna2024WebDev committed:
added useEffect in ThoughtList to be able to Sync updatedLikes with the latest message when they change

Anna2024WebDev committed:
added a lot of styling to the message-box, hearts, hoover effects and added time stamp

Anna2024WebDev committed:
more styling to thought-list, thought-message box and button

Anna2024WebDev committed:
added margin to center items

Anna2024WebDev committed:
added responsiveness

Anna2024WebDev committed:
fixed styling

Anna2024WebDev committed:
added accessibility to the page such as adding aria labels for the form and setup the elements in correct order and changed color contrast

Anna2024WebDev committed:
changed if/else statement for timeDifference so it shows days instead of hours after 24 hours has passed

Anna2024WebDev committed:
updated title

Anna2024WebDev committed:
changed font color of header to fix contrast

Anna2024WebDev committed:
added viewport meta tag and increased font size to fix the problem with screen zooming in on mobile

Anna2024WebDev committed:
more styling

Anna2024WebDev committed:
changed the <meta name viewport and maximum-scale so it fits the accessibility measures

Anna2024WebDev committed:
fixed styling

Anna2024WebDev committed:
added some extra comments to explain the code as well as an icon

Anna2024WebDev committed:
added hover effect to button

Anna2024WebDev committed:
removed redundant code and added more comments to the code

HIPPIEKICK commented on src/Styles/App.css:
```diff
@@ -0,0 +1,61 @@
+    max-width: 95%;
+    padding: 10px;
+    box-sizing: border-box;
+    resize: none;
```
⭐️

HIPPIEKICK reviewed (changes requested):
Good job with your happy thoughts app Anna! You’ve successfully copied the design as well as fetched from an API in React 🥳

Really good error handling as well as structure of your components. Just remember that your folders should be in lowercase (unless you structure your files so that you have one folder called e.g. Header with a Header.jsx and a Header.css in it). A next step for better structure would be to break it down even further, e.g. abstracting the API calls and helper functions such as your timeAgo function (kudos on coding that from scratch by the way!)

I can see that you put some thought into accessibility as well - great!

Overall, your code is easy to follow but I would like to add some notes on clean code:
Be consistent with single or double quotes and whether or not you’re using semicolons
Remove unused code
Order your imports in the following order:
```
// React and core libraries
import { useState, useEffect } from 'react';

// Components
import { Header } from"./Components/Header";
import { ThoughtForm } from "./Components/ThoughtForm";
import { ThoughtList } from"./Components/ThoughtList";

// Assets (CSS, images)
import "./Styles/App.css";
```

**Changes requested**
- Clean up code

Anna2024WebDev committed:
added more comments to explain the code

Anna2024WebDev committed:
added more comments to the code

Anna2024WebDev committed:
removed unused code as well as semicolons and consistent in using double quotes, ordered the components in correct order and changeed some styling

Anna2024WebDev committed:
small fix

Anna2024WebDev commented:
Thank you for your great feedback and comments @HIPPIEKICK! 

I have now;

1. Removed semicolons since I prefer not to use it if not necessary (such in CSS)
2. Use double quotations everywhere I can. 
3. Removed unused code such as API requests for likes to two components. I had added it both in the ThoughtList component as well as the ThoughtHeart component. It was removed from the ThoughtList since I have the like changes in the HeartComponent. 
4. Rearranged the components in correct order and changed the names of the folders to lowercase. 

Can you please have a look again?

ps. The timeAgo function I got help from chatGPT with, maybe I was not clear enough in the readme file about that. 🙈

Anna2024WebDev committed:
tried to push again because of problems with deploy update in netlify

Anna2024WebDev committed:
try fixing problem with deploy

Anna2024WebDev committed:
try pushing again

Anna2024WebDev committed:
tried changing name of Header component and pushing again

Anna2024WebDev committed:
new try

Anna2024WebDev committed:
temporary renamed folders

Anna2024WebDev committed:
temp name to komponenter

Anna2024WebDev committed:
commit again to change name on component folder

Anna2024WebDev committed:
updated h1 font size for mobile

JoyceKuode commented on src/App.jsx:
```diff
@@ -1,3 +1,71 @@
+    <div className="App">
+
+      {isLoading && <h1>Loading...</h1>} {/* Show loading message while data is being fetched */}
+      {/* Render the app content once loading is complete */}
```
Nice job with showing the loading message ⭐

JoyceKuode commented on src/assets/smilingheart.png:
What a cute icon ! 😊🩷

JoyceKuode commented on src/components/ThoughtForm.jsx:
```diff
@@ -0,0 +1,84 @@
+      <h2>What is making you happy right now?</h2>
+      <form onSubmit={handleSubmit}>
+        {/* Label for accessibility (hidden visually but helps screen readers) */}
+        <label htmlFor="thoughtText" className="visually-hidden">Share your happy thought:</label>
```
Nice with the extra attention to a11y 👏

JoyceKuode commented on src/components/ThoughtHeart.jsx:
```diff
@@ -0,0 +1,51 @@
+    <span
+      className={`heart-count ${liked ? "liked" : ''}`}
+      onClick={handleLike}
+      style={{ cursor: "pointer" }}>
```
I think it's generally considered best practice to put styling in CSS for separation of concerns, reusability and maintainability. You have declared a className here which you can use to add style in the ThoughtHeart.css file

JoyceKuode commented on src/styles/ThoughtHeart.css:
```diff
@@ -0,0 +1,22 @@
+  border-radius: 50%;
+  padding: 10px;
+  margin-right: 5px;
+}
```
I think the heart-count style I mentioned in the ThoughtHeart.jsx file can be used here, and you can replace span with .heart-count. Using a specific class that you already created rather than the generic span selector can improve maintainability and clarity of your code. Something like this maybe ? 

```
.heart-count {
  background-color: #EAEAEA;
  border-radius: 50%;
  padding: 10px;
  margin-right: 5px;
/* Add cursor: pointer here */
  cursor: pointer; 
}

.heart-count.liked {
  /* Add styles for the liked state */
}
```

JoyceKuode commented on src/App.jsx:
```diff
@@ -1,3 +1,71 @@
+// ThoughtList Component: Responsible for displaying a list of thoughts passed down from the App component.
+
+// Each component has it's separate CSS file*/ }
+
```
Thank you for this very clear explanation of your components. 🙏

JoyceKuode commented on src/styles/App.css:
```diff
@@ -0,0 +1,62 @@
+    max-width: 95%;
+    /* Use a smaller max-width on mobile */
+    padding: 10px;
+  }
```
You have styles here for .thought-form and .thought-list, but you also have ThoughtForm.css and ThoughtList.css files... You can also create a media query in the other CSS files. Consolidating all styles for the .thought-form class into the ThoughtForm.css file and the .thought-list class into the ThoughtList.css file will improve consistency and avoid conflicts that can lead to unexpected behavior.

JoyceKuode commented on src/styles/ThoughtForm.css:
```diff
@@ -0,0 +1,52 @@
+
+.visually-hidden {
+  display: none;
+}
```
An example of the comment I made in your App.css file : 

You can move the media query from the App.css over here like this (and similarly create a media query in your ThoughtList.css for the .thought-list class), and that way changes you might make to a class will be applied consistently across your project (plus for larger projects, it makes it easier to find in a single file 🔦): 

```
@media (max-width: 600px) {
  .thought-form {
    max-width: 95%;
    padding: 10px;
  }
}
```

JoyceKuode reviewed (commented):
I was really impressed overall with your happy thoughts app, Anna! 🩷 You did a great job hitting all the requirements and matching the design but with your own flair. Your code looks really organized and you have such clarity with your comments so I appreciate being able to follow along easily. I just had a couple of suggestions to deal with separation of concerns and maintainability. Hope this helps as our projects get larger and more complex.

Keep up the fantastic work, gold star for you ! 🌟 I'm excited to see what you'll create next 😊

Anna2024WebDev committed:
replaced API with my own MongoDB API

Anna2024WebDev committed:
changed the order of is loading text

Anna2024WebDev committed:
added styling for loading text

Anna2024WebDev committed:
removed the minimum loading time for loading state

Anna2024WebDev committed:
added timeout to loading text

Anna2024WebDev committed:
removed redundant code

Anna2024WebDev committed:
updated components to optimize time load and error handling

Anna2024WebDev committed:
debugging
